File: find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    int solve(vector<int>& a, int x){
         int n=a.size();
         int first_pos=n;
         int low=0, high=n-1;
         while(low<=high){
             int mid=low+(high-low)/2;
             if(a[mid]>=x){
                 first_pos=mid;
                 high=mid-1;
             }
             else
                 low=mid+1;
         }
        return first_pos;
    }
     vector<int> searchRange(vector<int>& nums, int target) {
         int first=solve(nums,target);
         int last=solve(nums,target+1)-1;
         if(first<=last)
             return {first,last};
         return {-1,-1};
    }
};

File: find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
	def searchRange(self, lst: List[int], t: int) -> List[int]:
		left = 0
		right = len(lst) - 1
		temp = -1

		while left <= right:

			mid = left + (right - left) // 2

			if lst[mid] == t:
				temp = mid
				right = mid - 1

			elif lst[mid] > t:
				right = mid - 1

			else:
				left = mid + 1


		temp1 = -1
		left, right = 0, len(lst) - 1

		while left <= right:

			mid = left + (right - left) // 2

			if lst[mid] == t:
				temp1 = mid
				left = mid + 1

			elif lst[mid] > t:
				right = mid - 1
			else:
				left = mid + 1

		return [temp, temp1]